### PR TITLE
Refactors delete comment

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9198

## Description of Changes
- Refactors old endpoint
- Refactors bot integration

## Test Plan
- Can delete comments 
- Can delete bot comments

## Deployment Plan
- Deploy discord bot

### CODE REVIEW

File: packages/commonwealth/client/scripts/state/api/comments/deleteComment.ts [hunk 1]
```diff
@@ -75,18 +29,33 @@ const useDeleteCommentMutation = ({
...
   const { checkForSessionKeyRevalidationErrors } = useAuthModalStore();
 
-  return useMutation({
-    mutationFn: deleteComment,
+  return trpc.comment.deleteComment.useMutation({
     onSuccess: async (response) => {
+      // Important: we render comments in a tree, if the deleted comment is a
+      // leaf node, remove it, but if it has replies, then preserve it with
+      // [deleted] msg.
+      const canvasSignedData = await signDeleteComment(
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/comments/deleteComment.ts:37]: shouldn't this canvas-signed data be included in the payload (similar to the old hanlder)?

**Review by mzparacha — COMMENTED** [review]
Thread and comment creation is failing for me

image

Couldn't try delete comment (not able to create one to delete)

File: packages/commonwealth/client/scripts/state/api/comments/deleteComment.ts (hunk 1 above)
Comment (Rotorsoft) [reply packages/commonwealth/client/scripts/state/api/comments/deleteComment.ts:37]: canvas data is not being used in the command anywhere, that's why I decided to include it in the response instead... so caller of the mutation doesn't need to provide anything else other than the comment id... that's the main change I wanted you to review, let me know if that works

**Comment by Rotorsoft** [comment]
> Thread and comment creation is failing for me
> 
> image
> Couldn't try delete comment (not able to create one to delete)

I noticed some changes to the canvas interface I had to merge in this PR, mainly canvas hash was replaced with canvas message id... not sure if this is related to that... will look on my end

File: libs/model/src/comment/DeleteComment.command.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            model: models.Thread,
+            attributes: ['community_id'],
+            required: true,
+          },
+        ],
+        logging: true,
+      });
+      mustExist('Comment', comment);
+
+      if (comment.address_id !== address!.id && address.role === 'member')
```
Comment (timolegros) [inline libs/model/src/comment/DeleteComment.command.ts:33]: Shouldn't this be handled in `isAuthorized`? If this is handled in `isAuthorized` then we can also remove line 15 `mustBeAuthorized` from above.

File: libs/model/src/comment/DeleteComment.command.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { isAuthorized, type AuthContext } from '../middleware';
+import { mustBeAuthorized, mustExist } from '../middleware/guards';
+
+export function DeleteComment(): Command<
+  typeof schemas.DeleteComment,
+  AuthContext
+> {
+  return {
+    ...schemas.DeleteComment,
+    auth: [isAuthorized({})],
```
Comment (timolegros) [inline libs/model/src/comment/DeleteComment.command.ts:13]: Should this have `author: true`?

File: libs/model/test/thread/thread-lifecycle.spec.ts [hunk 4]
```diff
@@ -590,6 +592,38 @@ describe('Thread lifecycle', () => {
...
+    it('should throw delete when user is not author', async () => {
+      await expect(
+        command(DeleteComment(), {
+          actor: actors.rejected,
+          payload: {
+            comment_id: comment!.id,
+          },
+        }),
+      ).rejects.toThrowError(InvalidActor);
+    });
```
Comment (timolegros) [inline libs/model/test/thread/thread-lifecycle.spec.ts:626]: Should we add a quick test for deleting comments as an admin?

**Review by timolegros — CHANGES_REQUESTED** [review]
The API to delete comments works but the UI doesn't remove the comment until I refresh. Additionally, the following errors are logged in the client console when deleting a comment:
image

File: libs/model/src/comment/DeleteComment.command.ts (hunk 2 above)
Comment (Rotorsoft) [reply libs/model/src/comment/DeleteComment.command.ts:33]: That's the ideal way to handle it, with the new actor flag, etc... but it breaks the discord bot integration.
So, I'm keeping the existing pattern of loading inside the body when discord bot shares the command. 

Once we release the API, we can think about how to optimize this pattern ... maybe by finding the entity_id in the bot layer before the command gets invoked, and we can clean all at once.

File: libs/model/test/thread/thread-lifecycle.spec.ts (hunk 4 above)
Comment (Rotorsoft) [reply libs/model/test/thread/thread-lifecycle.spec.ts:626]: sure, will do

**Review by rbennettcw — APPROVED** [review]

**Comment by mzparacha** [comment]
Thread/comment creation/deletion/voting works for me but when editing a thread/comment, version history selector is removed and it only shows the date-time of latest edit.

cc: @Rotorsoft

**Comment by Rotorsoft** [comment]
> Thread/comment creation/deletion/voting works for me but when editing a thread/comment, version history selector is removed and it only shows the date-time of latest edit.
> 
> cc: @Rotorsoft

This is an issue in the schemas... .Thread and Comment are missing version histories field... thinking about fixing it in #9256 to avoid making changes in 3 open PRs... I've been merging all open PRs to move faster, but when issues are found in reviews it creates a lot of extra work 🤯

**Comment by Rotorsoft** [comment]
> The issue with version history exists on master as well.

yep, fixed in https://github.com/hicommonwealth/commonwealth/pull/9256

**Pull request merged**